src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java

```diff
@@ -0,0 +1,502 @@
+        conn.addRequestProperty("Content-Type", "text/turtle");
+        conn.setDoOutput(true);
+
+        // Generate random IRI, FDP will replace it with really unique
+        String uri = fdpBaseURI + "/fdp/catalog/" + UUID.randomUUID();
+        catalogDTO.setIri(uri);
+
+        CatalogMetadata catalogMetadata = CatalogTransformerUtils.dto2Metadata(catalogDTO);
+        catalogMetadata.setIdentifier(MetadataTransformerUtils.createIdentifier(uri));
+        MetadataTransformerUtils.setTimestamps(catalogMetadata);
+        String metadata = MetadataUtils.getString(catalogMetadata, RDFFormat.TURTLE);
+
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+        bw.write(metadata);
+        bw.flush();
+        bw.close();
+
+        if(conn.getResponseCode() == HttpURLConnection.HTTP_CREATED) {
+            String actualURI = conn.getHeaderField("Location");
+            CatalogMetadataParser catalogMetadataParser = new CatalogMetadataParser();
+
+            CatalogMetadata returnCatalogMetadata = catalogMetadataParser.parse(
+                    parseStatements(conn, actualURI),
+                    SimpleValueFactory.getInstance().createIRI(actualURI)
+            );
+            return CatalogTransformerUtils.metadata2DTO(returnCatalogMetadata);
+        } else {
+            throw new FairDataPointException(conn.getResponseCode(), conn.getResponseMessage());
+        }
+    }
+
+    /**
+     * Create a new dataset in FDP
+     *
+     * @param datasetDTO dataset to be created
+     * @return created dataset
+     * @throws IOException in case of a communication error
+     * @throws FairDataPointException in case that FDP responds with an unexpected code
+     */
+    public DatasetDTO postDataset(DatasetDTO datasetDTO) throws IOException, MetadataException, FairDataPointException {
+        HttpURLConnection conn = createConnection(fdpBaseURI + DATASET_PART, "POST", "text/turtle");
+        conn.addRequestProperty("Content-Type", "text/turtle");
+        conn.setDoOutput(true);
+
+        // Generate random IRI, FDP will replace it with really unique
+        String uri = fdpBaseURI + "/fdp/dataset/" + UUID.randomUUID();
+        datasetDTO.setIri(uri);
+
+        DatasetMetadata datasetMetadata = DatasetTransformerUtils.dto2Metadata(datasetDTO);
+        datasetMetadata.setIdentifier(MetadataTransformerUtils.createIdentifier(uri));
+        MetadataTransformerUtils.setTimestamps(datasetMetadata);
+        String metadata = MetadataUtils.getString(datasetMetadata, RDFFormat.TURTLE);
+
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+        bw.write(metadata);
+        bw.flush();
+        bw.close();
+
+        if(conn.getResponseCode() == HttpURLConnection.HTTP_CREATED) {
+            String actualURI = conn.getHeaderField("Location");
```

Comment:
Do you have any constants on the classpath to re-use for header names? Guava, or a commons lib?